ipydatagrid/datagrid.py

```diff
@@ -168,7 +168,7 @@ def _get_num_rows(self):
 def _data_to_json(x, obj):
     if isinstance(x, dict):
         return {str(k): _data_to_json(v, obj) for k, v in x.items()}
-    elif isinstance(x, (list, tuple)):
+    elif isinstance(x, (list, tuple, np.ndarray)):
```

Comment:
Hi @martinRenou, thanks for the review.

Did you mean something like:
```suggestion
    elif isinstance(x, np.ndarray):
        return _data_to_json(x.tolist(), obj)
    elif isinstance(x, (list, tuple)):
```

I assume that I still need to go through the Python list comprehension to serialise the values appropriately.

If yes, that gives me an improvement over the original when tried locally:

```
%%timeit
_data_to_json_comp(test, None)

5.64 s ± 3.74 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```
vs
```
%%timeit
_data_to_json_with_to_list(test, None)

5.01 s ± 5.53 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

where `test` is a 100x100x100 array.

Is that what you had in mind?

Reply:
I was more suggesting something like:

```python
elif isinstance(x, np.ndarray):
    return x.tolist()
```

> to serialise the leaf values appropriately

Indeed. My suggestion would only be valid for simple types: when the NumPy array to-list conversion becomes a valid JSON.

Reply:
> Alternatively, I can extract the leaf value conversion in a separate function that I can vectorise which is slightly faster

Yes, that would be the proper way if we consider serializing NumPy arrays of more fancy types.

Reply:
I realised that NumPy arrays can also contain dictionaries that we would have to serialise properly which means that I would have to apply the whole `_data_to_json` function. This was slower when vectorised using `np.vectorise` but using `.tolist()` as suggested was faster than the comprehension directly on the NumPy array so I used this.

I also simplified the function a bit and split the cases for serialising `int`s and `float`s to avoid unneeded checks on the `int` side.

Reply:
Hi @martinRenou, are you okay with this approach?